backend/systematic_indentation_fix.py

```
@@ -0,0 +1,211 @@
+        fixed_lines.append(line)
+        
+        # If this line ends with ':' and next line isn't properly indented
+        if line.strip().endswith(':') and i < len(lines) - 1:
+            next_line = lines[i + 1] if i + 1 < len(lines) else ""
+            current_indent = len(line) - len(line.lstrip())
+            next_indent = len(next_line) - len(next_line.lstrip()) if next_line.strip() else 0
+            
+            # If next line should be indented but isn't
+            if next_line.strip() and next_indent <= current_indent:
+                # Add a pass statement if next line doesn't seem to belong to this block
+                if not any(keyword in next_line for keyword in ['return', 'if', 'else', 'try', 'except', 'finally']):
```

Ruff (0.12.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)